test/unit/mocks/sender_link.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+var SenderLink = require('../../../lib/sender_link'),
+    putils = require('../../../lib/policies/policy_utilities'),
+    util = require('util');
+
+function MockSenderLink(session, options) {
+  MockSenderLink.super_.call(this, session, null, {
+    encoder: function(body) { return body; },
+    callback: putils.SenderCallbackPolicies.OnSent
+  });
+
+  this._created = 0;
+  this.session = session;
+  this.options = options;
+  this._clearState();
+}
+
+util.inherits(MockSenderLink, SenderLink);
+
+MockSenderLink.prototype.canSend = function() {
+  this.emit('canSend-called', this);
+  return this.capacity > 0;
```

Comment:
I see what you mean re: capacity. I was only using this because it was easier (for testing) than tracking session/link credits - we might be able to get away with using those now and just calling base.canSend here.